loader/src/sources/vts/geo.js

```diff
@@ -0,0 +1,164 @@
+    getStoreName: (data) => {
+      const m = data.name.match(/CVS Pharmacy, Inc. #?(\d+)/i);
+      if (m) {
+        return `CVS #${parseInt(m[1], 10)}`; // format as in cvs/api.js
+      }
+      return data.name;
+    },
+  },
+};
+
+function warn(message, context) {
+  console.warn(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Info);
+}
+
+function error(message, context) {
+  console.error(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Error);
+}
+
+async function getStores() {
+  try {
+    const response = await got({
+      url: `https://univaf-data-snapshots.s3.us-west-2.amazonaws.com/vts/vts-final-output-locations.geojson`,
+    });
+    return JSON.parse(response.body);
+  } catch (e) {
+    error(`Error fetching stored Vaccine the States data`, e);
+    return [];
+  }
+}
+
+/**
+ * Filter out some stores with bad data.
+ * @param {any} store
+ * @returns {boolean}
+ */
+function hasUsefulData(store) {
+  return (
+    store.properties.concordances?.length &&
+    store.geometry?.coordinates &&
+    store.properties.provider?.name &&
+    store.properties.provider?.name in dataProviders
+  );
+}
+
+function formatStore(store) {
+  const data = store.properties;
+
+  let result;
+  Sentry.withScope((scope) => {
+    const dataProvider = dataProviders[data.provider.name];
+    if (!dataProvider) {
+      error(`Unexpected provider name ${data.provider.name}`);
+      return null;
+    }
+
+    const provider = dataProvider.provider;
+    const name = dataProvider.getStoreName(data);
+
+    scope.setContext("location", {
+      id: store.id,
+      url: data.vts_url,
+      name: name,
+      provider: provider,
+    });
+
+    function splitConcordance(concordance) {
+      const colon = concordance.indexOf(":");
+      return [
+        concordance.substring(0, colon),
+        concordance.substring(colon + 1),
+      ];
+    }
+
+    function filterConcordancePairs(pairs) {
+      return pairs
+        .filter(([systemCode, value]) => {
+          const system = systemsToSend[systemCode];
+          if (!system) {
+            return null;
+          }
+
+          if (system == "rite_aid" && value.match(/^1\d{5}$/)) {
+            // Stores like Rite Aid in this data set may have a store number like 105612.
+            // IDs with that shape (6 digits, starting with '1') are likely misparsed from VTrckS.
+            return null;
+          }
+
+          return [system, value];
+        })
+        .filter(Boolean);
+    }
+
+    const systemsToSend = {
+      cvs: "cvs",
+      rite_aid: "rite_aid",
+      vaccinespotter_org: "vaccinespotter",
+      vaccinefinder_org: "vaccines_gov",
+      vaccinefinder: "vaccines_gov",
+    };
+
+    const concordances = data.concordances.map(splitConcordance);
+    const externalIds = filterConcordancePairs(concordances);
+    const univafPair = concordances.filter((v) => v[0] == "getmyvax_org")[0];
```

Comment:
Are we always guaranteed to have at least one element after the filter?

Reply:
We are not! If there isn't a concordance for `getmyvax_org` in the list, `univafPair` will end up `undefined`, which we guard against below.